Sources/CNIOLinux/shim.c

```diff
@@ -215,4 +215,17 @@ const int CNIOLinux_AT_EMPTY_PATH = AT_EMPTY_PATH;
 const unsigned long CNIOLinux_UTIME_OMIT = UTIME_OMIT;
 const unsigned long CNIOLinux_UTIME_NOW = UTIME_NOW;
 
+
+bool CNIOLinux_udp_max_segments_defined() {
+    #ifdef UDP_MAX_SEGMENTS
+    return true;
+    #else
+    return false;
+    #endif
+}
+
+#ifdef UDP_MAX_SEGMENTS
+const unsigned long CNIOLinux_UDP_MAX_SEGMENTS = UDP_MAX_SEGMENTS;
+#endif
+const unsigned long CNIOLinux_UDP_MAX_SEGMENTS = 0;
```

Comment:
Rather than have `max_segments_defined`, we should probably just return a sentinel value. `-1` is good.

Reply:
I normally try to avoid sentinel values as a rule but given we know that this value should comfortably fit within an `Int64` I can live with it here, particularly if it might help the optimizer see through this logic.